mp/src/game/client/momentum/ui/HUD/hud_mapfinished.cpp

```diff
@@ -446,6 +486,50 @@ void CHudMapFinishedDialog::SetCurrentPage(int pageNum)
     m_pZoneVelMax->SetText(CConstructLocalizedString(m_pwVelMax, m_pRunStats ? m_pRunStats->GetZoneVelocityMax(m_iCurrentPage, velType) : 0.0f));
 }
 
+void CHudMapFinishedDialog::LoadPlayerBestTime()
+{
+    const auto pPlayer = C_MomentumPlayer::GetLocalMomPlayer();
+    if (!pPlayer)
+    {
+        return;
+    }
+    const auto pRunData = pPlayer->GetCurrentUIEntData();
+
+    CMomReplayBase *pBase = MomUtil::GetBestTime(g_pGameRules->MapName(), pRunData->m_flTickRate,
+                                                 pRunData->m_iCurrentTrack, pRunData->m_iRunFlags);
+
+    if (!pBase)
+    {
+        return;
+    }
+    
+    m_pPbRunStats->FullyCopyFrom(*pBase->GetRunStats());
+    m_flPbOverallTime = pBase->GetRunTime();
+
+    delete pBase;
+}
+
+void CHudMapFinishedDialog::GetDiffString(float diff, char* compareStringOut, Color* compareColorOut) 
+{
+    char tempANSITimeOutput[BUFSIZETIME];
+    char diffSign;
+
+    diffSign = diff > 0.0f ? '+' : '-';
+
+    if (diffSign == '+')
+    {
+        compareColorOut->SetRawColor(m_cLoss.GetRawColor());
+    }
+    else
+    {
+        compareColorOut->SetRawColor(m_cGain.GetRawColor());
+    }
```

Comment:
```suggestion
    compareColorOut->SetRawColor(diff > 0.0f ? m_cLoss.GetRawColor() : m_cGain.GetRawColor());
```

Reply:
Though, what happens when the player ties their PB?

Reply:
[Treating this in the new version](https://github.com/momentum-mod/game/pull/705/commits/13de0cb7711f374791829004311a02806dae1b89#diff-af494e5a9a7c8487930ea44f9e53503dR502)